cmd/devstream/completion.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+// completionCMD represents the completion command
+func completionCMD(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "completion",
+		Short:                 "Generate the autocompletion script for dtm for the specified shell.",
+		Long:                  "See each sub-command's help for details on how to use the generated script.",
+		DisableFlagsInUseLine: true,
+		Args:                  cobra.ExactValidArgs(1),
+	}
+
+	binaryName := filepath.Base(os.Args[0])
+	bash := &cobra.Command{
+		Use:     "bash",
+		Short:   "generate autocompletion script for bash",
+		Example: bashExample(binaryName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return completionBash(out, cmd)
+		},
+	}
+
+	zsh := &cobra.Command{
+		Use:     "zsh",
+		Short:   "generate autocompletion script for zsh",
+		Example: zshExample(binaryName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return completionZsh(out, cmd)
+		},
+	}
+
+	fish := &cobra.Command{
+		Use:     "fish",
+		Short:   "generate autocompletion script for fish",
+		Example: fishExample(binaryName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return cmd.Root().GenFishCompletion(out, true)
+		},
+	}
+
+	powershell := &cobra.Command{
+		Use:     "powershell",
+		Short:   "generate autocompletion script for powershell",
+		Example: powershellExample(binaryName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return cmd.Root().GenPowerShellCompletionWithDesc(out)
+		},
+	}
+	cmd.AddCommand(bash, zsh, fish, powershell)
+
+	return cmd
+}
+
+func completionBash(out io.Writer, cmd *cobra.Command) error {
```

Comment:
We follow the [Standard Go Project Layout](https://github.com/golang-standards/project-layout), and the [`cmd`](https://docs.devstream.io/en/latest/development/project-layout/#cmd) directory is better not to contain a lot of logic:

> Don't put a lot of code in the application directory. If you think the code can be imported and used in other projects, then it should live in the /pkg directory. If the code is not reusable or if you don't want others to reuse it, put that code in the /internal directory. You'll be surprised what others will do, so be explicit about your intentions!

Based on this, I suggest we move the code from line 63 to the end of this file to another place, like `internal/pkg/completion/xxx.go` (only an example/a suggestion.)